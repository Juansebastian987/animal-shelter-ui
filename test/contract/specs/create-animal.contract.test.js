import { provider } from "./init-pact";
import { AnimalController } from "../../../controllers";
import {Matchers} from "@pact-foundation/pact";

describe('Given An Animal Service', () =>{ 
    beforeAll(async ()=>{
        await provider.setup();
    });

   describe('When a request the animals create', () =>{
        beforeAll(async ()=>{
            await provider.addInteraction({
                state: 'there an animal',
                uponReceiving: 'a request to get the animal created',
                state:"has an animal",
                withRequest: {
                    method: 'POST',
                    path: '/animals',			
                },
                willRespondWith: {
                    status:201,
                    body: Matchers.eachLike({
                        breed: Matchers.like("Bendali"),
                        gender: Matchers.like("Female"),
                        vaccinated: Matchers.boolean(true),
                        name: Matchers.string("Other Animal")
                    }, {min: 1})
                }
            });
       });
        it("Then it should return the right data", async() => {
            const response = await AnimalController.list();
            expect(response.data).toMatchSnapshot();
            
            await provider.verify();
            
        });
    });
    afterAll(async ()=>{
        await provider.finalize();
    });
});