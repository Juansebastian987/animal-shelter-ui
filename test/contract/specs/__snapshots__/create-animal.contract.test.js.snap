// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given An Animal Service When a request the animals create Then it should return the right data 1`] = `
Object {
  "breed": "Bendali",
  "gender": "Female",
  "name": "Other Animal",
  "vaccinated": true,
}
`;
